daemon/src/setup_contract.rs

```diff
@@ -496,75 +497,13 @@ async fn build_own_cfd_transactions(
     Ok(own_cfd_txs)
 }
 
-#[allow(clippy::too_many_arguments)]
-pub async fn roll_over(
-    mut sink: impl Sink<RolloverMsg, Error = anyhow::Error> + Unpin,
-    mut stream: impl Stream<Item = RolloverMsg> + Unpin,
-    (oracle_pk, announcement): (schnorrsig::PublicKey, olivia::Announcement),
-    rollover_params: RolloverParams,
-    our_role: Role,
-    our_position: Position,
-    dlc: Dlc,
-    n_payouts: usize,
-    complete_fee: FeeFlow,
-) -> Result<Dlc> {
-    let (rev_sk, rev_pk) = keypair::new(&mut rand::thread_rng());
-    let (publish_sk, publish_pk) = keypair::new(&mut rand::thread_rng());
-
-    sink.send(RolloverMsg::Msg0(RolloverMsg0 {
-        revocation_pk: rev_pk,
-        publish_pk,
-    }))
-    .await
-    .context("Failed to send Msg0")?;
-
-    let msg0 = stream
-        .next()
-        .timeout(ROLLOVER_MSG_TIMEOUT)
-        .await
-        .with_context(|| format_expect_msg_within("Msg0", ROLLOVER_MSG_TIMEOUT))?
-        .context("Empty stream instead of Msg0")?
-        .try_into_msg0()?;
-
-    let (punish_maker, punish_taker, other_punish_params) = create_punish_params(
-        our_role,
-        dlc.identity,
-        dlc.identity_counterparty,
-        msg0,
-        rev_pk,
-        publish_pk,
-    );
-
-    let own_cfd_txs = create_own_cfd_transactions(
-        &dlc,
-        rollover_params,
-        &announcement,
-        oracle_pk,
-        our_role,
-        our_position,
-        n_payouts,
-        complete_fee,
-        (punish_maker, punish_taker),
-    )
-    .await?;
-
-    sink.send(RolloverMsg::Msg1(RolloverMsg1::from(own_cfd_txs.clone())))
-        .await
-        .context("Failed to send Msg1")?;
-
-    let msg1 = stream
-        .next()
-        .timeout(ROLLOVER_MSG_TIMEOUT)
-        .await
-        .with_context(|| format_expect_msg_within("Msg1", ROLLOVER_MSG_TIMEOUT))?
-        .context("Empty stream instead of Msg1")?
-        .try_into_msg1()?;
-
-    let lock_amount = dlc.taker_lock_amount + dlc.maker_lock_amount;
-
-    let ((maker_identity, maker_punish_params), (taker_identity, taker_punish_params)) =
-        (punish_maker, punish_taker);
-    let commit_desc = commit_descriptor(
+fn build_commit_descriptor(
```

Comment:
This function seems to be unnecessary as all you do is passing the parameters down.

Reply:
Readability is the only reason why it exists - initially I had it inlined but found the unnecessarily bloated :)